js/canvas.js

```diff
@@ -0,0 +1,367 @@
+      }
+    });
+  }
+
+  clear() {
+    const { width, height } = this.canvas;
+
+    this.ctx.clearRect(0, 0, width, height);
+  }
+
+  isRectangle_a_and_b_crossed(rectA, rectB) {
+    const topSideOfRectA = rectA.y;
+    const leftSideOfRectA = rectA.x;
+    const rightSideOfRectA = rectA.x + rectA.width;
+    const bottomSideOfRectA = rectA.y + rectA.height;
+    const topSideOfRectB = rectB.y;
+    const leftSideOfRectB = rectB.x;
+    const rightSideOfRectB = rectB.x + rectB.width;
+    const bottomSideOfRectB = rectB.y + rectB.height;
+
+    if (bottomSideOfRectA <= topSideOfRectB || rightSideOfRectA <= leftSideOfRectB) {
+      return false;
+    }
+
+    if (topSideOfRectA >= bottomSideOfRectB || rightSideOfRectB <= leftSideOfRectA) {
+      return false;
+    }
+
+    return true;
+  }
+
+  isRectangle_a_and_b_sticked(rectA, rectB) {
+    const topSideOfRectA = rectA.y - stickyDistance;
+    const leftSideOfRectA = rectA.x - stickyDistance;
+    const rightSideOfRectA = rectA.x + rectA.width + stickyDistance;
+    const bottomSideOfRectA = rectA.y + rectA.height + stickyDistance;
+    const topSideOfRectB = rectB.y - stickyDistance;
+    const leftSideOfRectB = rectB.x - stickyDistance;
+    const rightSideOfRectB = rectB.x + rectB.width + stickyDistance;
+    const bottomSideOfRectB = rectB.y + rectB.height + stickyDistance;
+
+    if (bottomSideOfRectA <= topSideOfRectB || rightSideOfRectA <= leftSideOfRectB) {
+      return false;
+    }
+
+    if (topSideOfRectA >= bottomSideOfRectB || rightSideOfRectB <= leftSideOfRectA) {
+      return false;
+    }
+
+    return true;
+  }
+
+  isCrossedRectangle(rectA) {
+    return arrayOfRectangles.reduce((acc, rectB) => {
+      if (rectA !== rectB && this.isRectangle_a_and_b_crossed(rectA, rectB)) acc = true;
+
+      return acc;
+    }, false);
+  }
+
+  isStickedRectangle(rectA) {
+    return arrayOfRectangles.reduce((acc, rectB) => {
+      if (rectA !== rectB && this.isRectangle_a_and_b_sticked(rectA, rectB)) acc = true;
+
+      return acc;
+    }, false);
+  }
+
+  setStickedPropToRectangles() {
+    arrayOfRectangles.forEach(rect => {
+      rect.isSticked = this.isStickedRectangle(rect);
+    });
+  }
+
+  setCrossedPropToRectangles() {
+    arrayOfRectangles.forEach(rect => {
+      rect.isCrossed = this.isCrossedRectangle(rect);
+    });
+  }
+
+  setColorPropToRectangles() {
+    arrayOfRectangles.forEach(rect => {
+      rect.fillColor = rect.isCrossed ? redColorInHex : startedRectangleFillColor;
+    });
+  }
+
+  setStickedSideToDraggingRectangle(draggingRect, standingRect) {
+    const draggingRectCenter = {
+      x: draggingRect.x + draggingRect.width / 2,
+      y: draggingRect.y + draggingRect.height / 2,
+    };
+    const standingRectCenter = {
+      x: standingRect.x + standingRect.width / 2,
+      y: standingRect.y + standingRect.height / 2,
+    };
+
+    if (draggingRectCenter.x < standingRectCenter.x && draggingRectCenter.y < standingRectCenter.y) {
```

Comment:
Lines 148 - 171 need to be simplify. Also I suggest don't use string constant as a base for logic